src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -349,9 +349,24 @@ void drawing_classifier::init_training(
   std::vector<std::string> gpu_names = training_compute_context_->gpu_names();
   print_training_device(gpu_names);
 
-  // Initialize the neural net. Note that this depends on statistics computed
-  // by the data iterator.
-  nn_spec_ = init_model();
+  // If warm_start enabled, an 'mlmodel_path' is passed in options.
+  // Else, call init_model() to initialize the neural net using xavier.
+  if (opts.find("mlmodel_path") != opts.end() ) {
+    // Initialize the neural net with warm start model weights.
+    auto mlmodel_path_iter = opts.find("mlmodel_path");
+    if (mlmodel_path_iter == opts.end()) {
```

Comment:
this logic is redundant.

Reply:
Yeah, we should just initialize `mlmodel_path_iter` first and test that in line 354